File: archetype/project/java/app/src/test/java/{{packagePath}}/{{className}}Test.java
package {{packageName}};

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class {{className}}Test {
    @Test
    void testGreet() {
        {{className}} app = new {{className}}();
        assertEquals("Hello, Test!", app.greet("Test"));
    }
}

File: archetype/module/kotlin/src/test/kotlin/{{packagePath}}/{{className}}Test.kt
package {{packageName}}

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class {{className}}Test : StringSpec({
    "greet should return greeting message" {
        val app = {{className}}()
        app.greet("Test") shouldBe "Hello, Test!"
    }
})